File: play-scala-websocket/app/controllers/HomeController.scala
package controllers

import javax.inject._
import actors.MyWebSocketActor
import akka.actor.ActorSystem
import akka.stream.Materializer
import play.api._
import play.api.libs.json.JsValue
import play.api.libs.streams.ActorFlow
import play.api.mvc._

/**
 * This controller creates an `Action` to handle HTTP requests to the
 * application's home page.
 */
@Singleton
class HomeController @Inject() (implicit system: ActorSystem, materializer: Materializer) extends Controller {

  /**
   * Create an Action to render an HTML page with a welcome message.
   * The configuration in the `routes` file means that this method
   * will be called when the application receives a `GET` request with
   * a path of `/`.
   */
  def index = Action {
    Ok(views.html.index("Your new application is ready."))
  }

  def ws = WebSocket.accept[String, String] { request => ActorFlow.actorRef(out => MyWebSocketActor.props(out))}

  // def ws = WebSocket.accept[JsValue, JsValue] { request => ActorFlow.actorRef(out => MyWebSocketActor.props(out))}

  def wspage = Action { Ok(views.html.wspage()) }
}

File: play-java/app/controllers/HomeController.java
package controllers;

import actors.MyWebSocketActor;
import akka.stream.javadsl.Flow;
import play.api.libs.streams.ActorFlow;
import play.mvc.*;

import views.html.*;

/**
 * This controller contains an action to handle HTTP requests
 * to the application's home page.
 */
public class HomeController extends Controller {

    /**
     * An action that renders an HTML page with a welcome message.
     * The configuration in the <code>routes</code> file means that
     * this method will be called when the application receives a
     * <code>GET</code> request with a path of <code>/</code>.
     */
    public Result index() {
        return ok(index.render("Your new application is ready."));
    }

    public LegacyWebSocket<String> ws() {
        return WebSocket.withActor(MyWebSocketActor::props);
    }

    public Result wspage() {
        return ok(wspage.render());
    }

    /* AkkaStream Java API is not ready yet!!!
    public WebSocket<String> wsnew() {
        return WebSocket.Text.accept(requestHeader -> ActorFlow.actorRef(MyWebSocketActor::props));
    }
    */

}
